InventoryVisualMerchandiser/Plugin/Model/Resolver/QuantityAndStockPlugin.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryVisualMerchandiser\Plugin\Model\Resolver;
+
+use Magento\Catalog\Model\ResourceModel\Product\Collection;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\InventoryCatalogApi\Api\DefaultStockProviderInterface;
+use Magento\InventoryIndexer\Model\StockIndexTableNameResolverInterface;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Api\StockResolverInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\VisualMerchandiser\Model\Resolver\QuantityAndStock;
```

Comment:
It's not acceptable to have Inventory related logic in the magento core. Please, remove `Magento\VisualMerchandiser\Model\Resolver\QuantityAndStock` and move logic to the inventory repo and handle additional cases mentioned in this PR.